interface/controller/pokemon_controller.go

```diff
@@ -0,0 +1,99 @@
+package controller
+
+import (
+	"net/http"
+	//"fmt"
+	//"errors"
+	"github.com/gorilla/mux"
+	"encoding/json"
+	"WizelineApi/domain/model"
+	"WizelineApi/service"
+	"strconv"
+)
+
+//Structura vacia del controlador. 
+type pokemonController struct{}
+
+type PokemonController interface{
+	GetAllPokemon()
+	GetPokemon()
+}
+
+
+
+func (pc *pokemonController) GetAllPokemons(res http.ResponseWriter, req *http.Request)  {
+	
+	pokemons := []model.Pokemon{}
+	err := service.ReadCSV("./test/bateria_csv/pokemon.csv", &pokemons)
+
+	if err != nil {
+		res.Header().Set("Content-Type", "application/json")
+		res.WriteHeader(http.StatusBadRequest)
+		res.Write([]byte(err.Error()))
+	}
+
+	//leer csv
+	// generar array del csv
+
+	res.Header().Set("Content-Type", "application/json")
+	res.WriteHeader(http.StatusOK)
+	json.NewEncoder(res).Encode(pokemons)
+}
+
+func (pc *pokemonController) GetPokemon(res http.ResponseWriter, req *http.Request) {
+	pokemons := []model.Pokemon{}
+	err := service.ReadCSV("./test/bateria_csv/pokemon.csv", &pokemons)
+
+	if err != nil {
+		res.Header().Set("Content-Type", "application/json")
+		res.WriteHeader(http.StatusBadRequest)
+		res.Write([]byte(err.Error()))
+	}	
+
+	pkm := model.Pokemon{}
+
+	params := mux.Vars(req)
+	id,err := strconv.Atoi(params["id"])
+	if err != nil {
+		res.Header().Set("Content-Type", "application/json")
+		res.WriteHeader(http.StatusBadRequest)
+		res.Write([]byte("Couldn't Parse properly to Integer."))
+		return
+	}
+
+	pkm, err = service.GetPokemonById(pokemons, id)
+	if err != nil {
+		res.Header().Set("Content-Type", "application/json")
+		res.WriteHeader(http.StatusOK)
+		res.Write([]byte(err.Error()))
+		return
+	}
```

Comment:
Return statements should be separated from previous code blocks.

Also, be careful with the information you are returning. These two cases handle errors, so you should use a proper HTTP status code. In the first case you are correctly returning a `400 Bad Request` status, but in the other one you returning a `200 OK` status.
```suggestion
	if err != nil {
		res.Header().Set("Content-Type", "application/json")
		res.WriteHeader(http.StatusBadRequest)
		res.Write([]byte("Couldn't Parse properly to Integer."))
		
		return
	}

	pkm, err = service.GetPokemonById(pokemons, id)
	if err != nil {
		res.Header().Set("Content-Type", "application/json")
		res.WriteHeader(http.StatusOK)
		res.Write([]byte(err.Error()))
		
		return
	}
```